File: coyote/models/kv_get_namespace_out.py
# this file is @generated
import typing as t
from pydantic import Field
from datetime import datetime

from ..internal.base_model import BaseModel

from .storage_type import StorageType


class KvGetNamespaceOut(BaseModel):
    created_at: datetime = Field(alias="created_at")

    max_storage_bytes: t.Optional[int] = Field(default=None, alias="max_storage_bytes")

    name: str

    storage_type: StorageType = Field(alias="storage_type")

    updated_at: datetime = Field(alias="updated_at")

File: src/models/kv_get_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::storage_type::StorageType;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct KvGetNamespaceOut {
    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_storage_bytes: Option<u64>,

    pub storage_type: StorageType,

    pub created_at: jiff::Timestamp,

    pub updated_at: jiff::Timestamp,
}

impl KvGetNamespaceOut {
    pub fn new(
        name: String,
        storage_type: StorageType,
        created_at: jiff::Timestamp,
        updated_at: jiff::Timestamp,
    ) -> Self {
        Self {
            name,
            max_storage_bytes: None,
            storage_type,
            created_at,
            updated_at,
        }
    }
}
